src/test/java/com/spring/core/BasketTest.java

```diff
@@ -0,0 +1,56 @@
+package com.spring.core;
+
+import com.spring.core.dao.impl.InMemoryBasketDaoImpl;
+import com.spring.core.model.Product;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class BasketTest {
```

Comment:
Only one negative test.
It's ugly

Reply:
Check last commit: 
1) I changed project structure:
now there is object with list of goods which creates when project start an you can add product from it to your basket
2) change some code and javaDoc that you said

*I'm adding more tests and read that you said